linux/LinuxProcessList.c

```diff
@@ -205,10 +209,12 @@ static void LinuxProcessList_updateCPUcount(ProcessList* super) {
       unsigned int max = MAXIMUM(existing, id + 1);
       if (max > currExisting) {
          this->cpuData = xReallocArray(this->cpuData, max + /* aggregate */ 1, sizeof(CPUData));
-         for (unsigned int j = currExisting; j < max; j++) {
-            this->cpuData[j].online = false;
+         if (currExisting == 0) {
+            memset(this->cpuData, '\0', (max + 1) * sizeof(CPUData)); /* zero everything */
+            this->cpuData[0].online = true; /* average is always "online" */
+         } else {
+            memset(this->cpuData + currExisting + 1, '\0', (max - currExisting) * sizeof(CPUData)); /* zero only added memory */
          }
```

Comment:
```suggestion
         if (currExisting == 0)
            memset(this->cpuData, '\0', sizeof(CPUData)); /* zero first entry (init) */
         memset(this->cpuData + currExisting + 1, '\0', (max - currExisting) * sizeof(CPUData)); /* zero added memory */
         this->cpuData[0].online = true; /* average is always "online" */
```

Doesn't hurt to have `this->cpuData[0].online` always be (re)set.